### PR TITLE
Revert trimming test workarounds

### LINKED ISSUE #57846: Remove trimming tests workaround for runtime bug
In https://github.com/dotnet/aspnetcore/pull/57742 we introduced a set of workarounds to the trimming test infrastructure to workaround an issue in the runtime repo.

Once https://github.com/dotnet/sdk/pull/43070 is merged we should revert the change made in https://github.com/dotnet/aspnetcore/pull/57742/commits/64f60ca87895f41a86b5121fbc3da5ea5efe17e1 and https://github.com/dotnet/aspnetcore/pull/57742/commits/7019edac0ce9f255dc65b0717d4a1f42d1b4e5fe.

### PR DESCRIPTION
**Pull request by MackinnonBuck**
Now that https://github.com/dotnet/sdk/pull/43070 is merged, we should be able to revert the linker workarounds applied in https://github.com/dotnet/aspnetcore/pull/57742.

Fixes #57846

### CODE REVIEW

**Comment by BrennanConroy** [comment]
We need to get the updated sdk first though?

**Comment by MackinnonBuck** [comment]
> We need to get the updated sdk first though?

Yep, I'll change the base of this PR to the SDK update PR when it gets opened.

File: src/Components/Forms/src/FieldIdentifier.cs [hunk 1]
```diff
@@ -126,7 +126,7 @@ private static void ParseAccessor<T>(Expression<Func<T>> accessor, out object mo
                         // so, given that it embeds captured values such as "this". We could consider special-casing
                         // for "() => something.Member" and building a cache keyed by "something.GetType()" with values
                         // of type Func<object, object> so we can cheaply map from "something" to "something.Member".
-                        var modelLambda = Expression.Lambda(typeof(Func<object?>), memberExpression.Expression);
+                        var modelLambda = Expression.Lambda(memberExpression.Expression);
```
Comment (BrennanConroy) [inline src/Components/Forms/src/FieldIdentifier.cs:129]:
```suggestion
                        var modelLambda = Expression.Lambda(typeof(Func<object?>), memberExpression.Expression);
```

File: src/Components/Forms/src/FieldIdentifier.cs [hunk 2]
```diff
@@ -201,7 +201,7 @@ static Func<object, object> CreateAccessor((Type model, MemberInfo member) arg)
     private static object GetModelFromIndexer(Expression methodCallExpression)
     {
         object model;
-        var methodCallObjectLambda = Expression.Lambda(typeof(Func<object?>), methodCallExpression!);
+        var methodCallObjectLambda = Expression.Lambda(methodCallExpression!);
```
Comment (BrennanConroy) [inline src/Components/Forms/src/FieldIdentifier.cs:204]:
```suggestion
        var methodCallObjectLambda = Expression.Lambda(typeof(Func<object?>), methodCallExpression!);
```

File: src/SignalR/server/Core/src/DynamicHub.cs [hunk 3]
```diff
@@ -1,13 +1,11 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System.Diagnostics.CodeAnalysis;
 namespace Microsoft.AspNetCore.SignalR;
```
Comment (BrennanConroy) [inline src/SignalR/server/Core/src/DynamicHub.cs:4]:
```suggestion
using System.Diagnostics.CodeAnalysis;

namespace Microsoft.AspNetCore.SignalR;
```

File: src/SignalR/server/Core/src/DynamicHub.cs [hunk 4]
```diff
@@ -1,13 +1,11 @@
...
 
 /// <summary>
 /// A base class for SignalR hubs that use <c>dynamic</c> to represent client invocations.
 /// </summary>
-[RequiresDynamicCode("DynamicHub requires dynamic code generation to construct a call site.")]
 public abstract class DynamicHub : Hub
```
Comment (BrennanConroy) [inline src/SignalR/server/Core/src/DynamicHub.cs:9]:
```suggestion
[RequiresDynamicCode("DynamicHub requires dynamic code generation to construct a call site.")]
public abstract class DynamicHub : Hub
```

File: src/SignalR/server/Core/src/DynamicHubClients.cs [hunk 5]
```diff
@@ -1,15 +1,13 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System.Diagnostics.CodeAnalysis;
 using Microsoft.AspNetCore.SignalR.Internal;
```
Comment (BrennanConroy) [inline src/SignalR/server/Core/src/DynamicHubClients.cs:4]:
```suggestion
using System.Diagnostics.CodeAnalysis;
using Microsoft.AspNetCore.SignalR.Internal;
```

File: src/SignalR/server/Core/src/DynamicHubClients.cs [hunk 6]
```diff
@@ -1,15 +1,13 @@
...
 
 namespace Microsoft.AspNetCore.SignalR;
 
 /// <summary>
 /// A class that provides <c>dynamic</c> access to connections, including the one that sent the current invocation.
 /// </summary>
-[RequiresDynamicCodeAttribute("DynamicHubClients requires dynamic code generation to construct a call site.")]
 public class DynamicHubClients
```
Comment (BrennanConroy) [inline src/SignalR/server/Core/src/DynamicHubClients.cs:11]:
```suggestion
[RequiresDynamicCodeAttribute("DynamicHubClients requires dynamic code generation to construct a call site.")]
public class DynamicHubClients
```

File: src/SignalR/server/Core/src/Internal/DynamicClientProxy.cs [hunk 7]
```diff
@@ -1,7 +1,6 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System.Diagnostics.CodeAnalysis;
 using System.Dynamic;
```
Comment (BrennanConroy) [inline src/SignalR/server/Core/src/Internal/DynamicClientProxy.cs:4]:
```suggestion
using System.Diagnostics.CodeAnalysis;
using System.Dynamic;
```

File: src/SignalR/server/Core/src/Internal/DynamicClientProxy.cs [hunk 8]
```diff
@@ -10,7 +9,6 @@ internal sealed class DynamicClientProxy : DynamicObject
 {
     private readonly IClientProxy _clientProxy;
 
-    [RequiresDynamicCodeAttribute("This constructor requires dynamic code generation to construct a call site.")]
     public DynamicClientProxy(IClientProxy clientProxy)
```
Comment (BrennanConroy) [inline src/SignalR/server/Core/src/Internal/DynamicClientProxy.cs:12]:
```suggestion
    [RequiresDynamicCodeAttribute("This constructor requires dynamic code generation to construct a call site.")]
    public DynamicClientProxy(IClientProxy clientProxy)
```

**Review by wtgodbe — APPROVED** [review]

**Pull request merged**